server/embed/config_test.go

```diff
@@ -1041,3 +1045,27 @@ func TestSetFeatureGatesFromExperimentalFlags(t *testing.T) {
 		})
 	}
 }
+
+func TestMatchNewConfigAddFlags(t *testing.T) {
+	cfg := NewConfig()
+	fs := flag.NewFlagSet("etcd", flag.ContinueOnError)
+	cfg.AddFlags(fs)
+	require.NoError(t, fs.Parse(nil))
+
+	newConfig := NewConfig()
+	// TODO: Remove the following assigments when both match.
+	newConfig.SelfSignedCertValidity = 1
+	newConfig.TlsMinVersion = string(tlsutil.TLSVersion12)
+	newConfig.DiscoveryCfg.Secure.InsecureTransport = true
+	newConfig.AutoCompactionRetention = "0"
+	newConfig.ExperimentalDistributedTracingAddress = "localhost:4317"
+	newConfig.ExperimentalDistributedTracingServiceName = "etcd"
+	newConfig.LogFormat = "json"
+	newConfig.ExperimentalTxnModeWriteWithSharedBuffer = true
+	// TODO: Reduce number of unexported fields set in config
+	if diff := cmp.Diff(newConfig, cfg, cmpopts.IgnoreUnexported(transport.TLSInfo{}, Config{}), cmp.Comparer(func(a, b featuregate.FeatureGate) bool {
```

Comment:
Are you going to resolve the TODO in this PR or in a followup PR?

Also I am a little confused why cfg and newConfg match even after you change some fields' value for newConfig? Why we do not see workflow failure? Did I miss anything?

Reply:
Yes, this PR shows the differences between the defaults and AddFlags, I plan to fix it in next PR.